### PR TITLE
Code generator code gardening

### PR DESCRIPTION
**Pull request by matthchr**
**What this PR does / why we need it**:
Fixes a few small issues which had come up in previous PRs but I had put off fixing in that PR due to the size of the change.

**Special notes for your reviewer**:
I believe @theunrepentantgeek had requested both of these at some point in the past, so tagging him.

### CODE REVIEW

File: hack/generator/pkg/astmodel/armconversion/shared.go [hunk 1]
```diff
@@ -41,8 +41,8 @@ func (builder conversionBuilder) propertyConversionHandler(
 	fromType *astmodel.ObjectType) []dst.Stmt {
 
 	for _, conversionHandler := range builder.propertyConversionHandlers {
-		stmts := conversionHandler(toProp, fromType)
-		if len(stmts) > 0 {
+		stmts, matched := conversionHandler(toProp, fromType)
+		if matched {
 			return stmts
 		}
```
Comment (theunrepentantgeek) [inline hack/generator/pkg/astmodel/armconversion/shared.go:47]: Nice. I think this is easier to grok now.

File: hack/generator/pkg/codegen/testdata/ArmResource/Arm_test_id_resource_reference_azure.golden [hunk 2]
```diff
@@ -324,6 +324,10 @@ func (fakeResourceProperties *FakeResourceProperties) PopulateFromARM(owner genr
 		return fmt.Errorf("unexpected type supplied for PopulateFromARM() function. Expected FakeResourcePropertiesARM, got %T", armInput)
 	}
 
+	// no assignment for property ‘Reference’:
+
+	// no assignment for property ‘SubnetReference’:
```
Comment (theunrepentantgeek) [inline hack/generator/pkg/codegen/testdata/ArmResource/Arm_test_id_resource_reference_azure.golden:329]: Really minor, but maybe these don't need the trailing `:` #trivial #nitpick

**Review by theunrepentantgeek — APPROVED** [review]
Nice.

gif

**Review by Porges — APPROVED** [review]

File: hack/generator/pkg/codegen/testdata/ArmResource/Arm_test_id_resource_reference_azure.golden (hunk 2 above)
Comment (matthchr) [reply hack/generator/pkg/codegen/testdata/ArmResource/Arm_test_id_resource_reference_azure.golden:329]: Fixed

**Pull request merged**